external-crates/move/crates/move-compiler/tests/linter/self_assignment.move

```diff
@@ -0,0 +1,16 @@
+module 0x42::M {
+
+    fun func1() {
+        let x = 5;
+        let y = 10;
```

Comment:
Could this be extended to handle, e.g., `let (x, y) = (x, y)` ?

Reply:
`let (x, y) = (x, y)` does not work in sui move. Tuple only appear in expressions (usually in the return position for a function)